docs/docs/dev_docs/contracts/compiling.md

```diff
@@ -1,22 +1,136 @@
 # Compiling contracts
 
-Please use the [TUTORIAL-TEMPLATE](../../TUTORIAL_TEMPLATE.md) for standalone guides / tutorials.
+Once you have written your [contracts](../contracts/main.md) in Noir, you will need to compile them into an artifact in order to use them. In this guide we will cover how to do so, both using the CLI and programmatically, as well as generating helper typescript and Noir interfaces for interacting with them.
 
-:::danger TODO
-TODO: this entire page
+## Prerequisites
+
+You will need the Noir build tool `nargo`, which you can install via [`noirup`](https://github.com/noir-lang/noirup). Make sure you install the `aztec` version of nargo:
+
+```bash
+curl -L https://raw.githubusercontent.com/noir-lang/noirup/main/install | bash
+noirup -v aztec
+```
+
+:::info
+You can re-run `noirup -v aztec` whenever you want to update to the latest version of Noir supported by the Aztec Network.
 :::
-## Compiling a Noir Contract
 
-You can use the `master` branch of Noir/nargo (hooray).
+## Compile using the CLI
+
+To compile using the Aztec CLI, first install it:
+
+`npm install -g @aztec/cli`
+
+Then run the `compile` command with the path to your Noir contract project:
+
+```
+aztec-cli compile ./src/noir
+```
+
+This will output a JSON artifact for each contract in the project to a `target` folder containing their [ABI](./abi.md), which you can use for deploying or interacting with your contracts.
+
+### Typescript
+
+You can use the compiler to autogenerate type-safe typescript classes for each of your contracts. These classes define type-safe methods for deploying and interacting with your contract based on their ABI.
+
+To generate them, include a `--typescript` option in the compile command with a path to the target folder for the typescript files:
+
+```
+aztec-cli compile --typescript ./src/ts/contracts ./src/noir
```

Comment:
Maybe make the input path name more descriptive (as per another comment).
We might even want to have required named cli parameters for the output-dir and input-dir? (But not for this PR).

Reply:
> We might even want to have required named cli parameters for the output-dir and input-dir?

Hmm disagree there. If there is a very specific single thing that is the target of a command, I think it makes more sense as a positional argument rather than a named one. In this case, "I want to compile this". As for output dir, I think it's better that it's optional, as long as we're telling the user where things were written to.

Reply:
It's interesting, because my intuition is that the ordering of the args (as they are ordered currently) is backwards. I would expect "the directory I'm compiling" to be the 1st argument, and "the directory I'm outputting code to" to be the 2nd argument. (Similar to the unix `mv` or `cp` commands).

It was this confusion (relative to my intuition) that led me to propose naming the arguments (to avoid the confusion).

Alternatively, we could copy `solc` and have a positional argument for the source directory, and an `--output` flag for the output directory?

Anyway, this could be a separate issue and PR (if at all). I don't want to hold up this lovely page, given that the current explanation accurately reflects the current code.

Reply:
> Alternatively, we could copy solc and have a positional argument for the source directory, and an --output flag for the output directory?

That's my proposal here. The source directory is the only thing required, and the main object you are referring to. The output directory is optional, so we can keep it behind an option.

Just to make sure it's clear: in this example `./src/ts/contracts` is **not** a positional argument, but the value of the `typescript` option.

Reply:
Ohhhh, I misunderstood and assumed --typescript was a flag without any arguments.